app/tests/test_search_utils.py

```diff
@@ -0,0 +1,167 @@
+from unittest.mock import patch
+
+import pytest
+
+from app.main.util.search_utils import (
+    build_dsl_query,
+    execute_search,
+    get_all_fields_excluding,
+    get_filtered_list,
+    get_pagination_info,
+    get_param,
+    get_query_and_search_area,
+    get_query_multi_match,
+    get_query_source_rules,
+    setup_opensearch,
+)
+from app.tests.test_search import MockOpenSearch
+
+
+def test_get_param(app):
+    with patch(
+        "flask.request.form.get", return_value="test_form_value"
+    ) as mock_form_get, patch(
+        "flask.request.args.get", return_value="test_arg_value"
+    ):
+
+        assert get_param("param") == "test_form_value"
+
+        mock_form_get.return_value = ""
+        assert get_param("param") == "test_arg_value"
+
+
+@patch(
+    "app.main.util.search_utils.get_param",
+    side_effect=["test_query", "test_area"],
+)
+def test_get_query_and_search_area(mock_get_param):
+    query, search_area = get_query_and_search_area()
+    assert query == "test_query"
+    assert search_area == "test_area"
+
+
+@patch("app.main.util.search_utils.OpenSearch", MockOpenSearch)
+def test_setup_opensearch(app):
+    with app.app_context():
+        client = setup_opensearch()
+        assert isinstance(client, MockOpenSearch)
```

Comment:
I think we can improve this by not mocking OpenSearch and changing to

`assert isinstance(client, OpenSearch)`
and can even assert the properties too